MBS-11239 Add GradleLoggerPlugin.kt (#1271)
avito-tech/avito-android: docs/content/contributing/Logging.md

# Logging

We have common libs those are reused in both environments: Android and Gradle. So we need a logger is not dependent on the concrete environment. \

> We create `com.avito.logger.Logger`. It helps us reuse code. 

Reviewer: ```suggestion
We've created `com.avito.logger.Logger`. It helps us to reuse code. 
```

## Logging in Gradle

> To obtain a logger for Gradle we create the `GradleLoggerPlugin`. It must be applied to the root project higher then others plugins dependent on it. \

Reviewer: ```suggestion
To obtain a logger for Gradle we create the `GradleLoggerPlugin`. It must be applied to the root project before others plugins dependent on it. \
```

You create `LoggerFactory` for a `project` or a `task`

```kotlin
val projectLoggerFactory = GradleLoggerPlugin.getLoggerFactory(project)
val taskLoggerFactory = GradleLoggerPlugin.getLoggerFactory(task)
```

### PrintlnLoggerHandler

> Gradle use lifecycle level by default, but to see info or debug level you have to set it for whole Gradle run

Reviewer: ```suggestion
Gradle uses lifecycle level by default, but to see info or debug level you have to set it for the whole Gradle run
```

via `--info` or `--debug`, which made console output unreadable and build slow.

There is an issue for
that: [gradle/#1010 Ability to set log level for specific task](https://github.com/gradle/gradle/issues/1010)

> `PrintlnLoggerHandler` add you ability to tune log level for avito plugins separately

Reviewer: ```suggestion
`PrintlnLoggerHandler` adds the ability to tune log level for avito plugins separately
```
